File: translations/codenet/uncommented/C++2Go/deepseek/Code/s447383972.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func readInt() int {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())
	a := make([]int, n)
	for i := 0; i < n; i++ {
		scanner.Scan()
		a[i], _ = strconv.Atoi(scanner.Text())
	}
	sort.Ints(a)
	q := make([]int, n)
	tot := 0
	for i := 0; i < n; i++ {
		if i < n-1 && a[i] == a[i+1] {
			q[tot] = a[i]
			tot++
			i++
		}
	}
	fmt.Println(q[tot-1] * q[tot-2])
}

File: translations/codenet/uncommented/C++2J/deepseek/Code/s447383972.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            a[i] = scanner.nextInt();
        }
        Arrays.sort(a, 1, n + 1);
        int[] q = new int[n + 1];
        int tot = 0;
        for (int i = 1; i <= n; i++) {
            if (a[i] == a[i + 1]) {
                q[++tot] = a[i];
                i++;
            }
        }
        System.out.println(q[tot] * q[tot - 1]);
    }
}

